torch/fx/experimental/proxy_tensor.py

```diff
@@ -170,6 +183,24 @@ def wrapped(*args):
     return wrapped
 
 
+class ProxyTorchDispatchMode(TorchDispatchMode):
+    def __init__(self, tracer):
+        self.tracer = tracer
+
+    def __torch_dispatch__(self, func_overload, types, args=(), kwargs=None):
+        func = func_overload.overloadpacket
+        if any(tuple(isinstance(arg, ProxyTensor) for arg in args)):
+            return func(*args, **kwargs)  # mode is disabled, this redispatches to ProxyTensor's torch dispatch
+        else:
+            proxy_out = self.tracer.create_proxy('call_function', func, args, kwargs,
+                                     name=self.tracer.graph._target_to_str(func.__name__))
+
+            with no_dispatch():
+                real_out = func_overload(*args, **kwargs)
+
+            return wrap_output(real_out, proxy_out)
+
+
 def make_fx(f, decomposition_table=None):
```

Comment:
In the original hacky PR, I also added a flag to `make_fx` that would let the user choose whether or not to trace the factory functions. Even as I was doing it, it felt like it added a lot of branching without a clear use case, so I took it out of this PR and just turned on tracing factory functions by default but happy to hide the behavior behind a flag if we prefer

cc @Chillee @anijain2305

Reply:
Is it a lot of branching? Don't we just add a branch here? https://github.com/pytorch/pytorch/blob/0ab4ded6c8b43b83d3b7633e653fef75a19306ca/torch/fx/experimental/proxy_tensor.py#L148

Reply:
Hmm fair point. I guess I was branching to avoid the extra no_dispatch calls but we probably don't have to do that

Still wanting to check, is there a use case for having a flag instead of just having it on all the time?

Reply:
Yes, definitely initially. If things work out then in the long term we can just have it on all the time. But for now, I think it makes sense to have a flag so folks can opt into the old behavior.